### ISSUE TITLE
Gfx issue flying dragons

### ISSUE THREAD

**Issue by ercan1441** [MiSTer-devel/Arcade-Tecmo_MiSTer#8]
In first stage the flying 'dragons' get split quite often.
Check especially between 5-8 sec. in attached video shot.

Using the latest Tecmo_20200810.rbf core.

https://user-images.githubusercontent.com/5685798/105010631-43628600-5a3c-11eb-9712-f8cfac0aab31.mp4

**Comment by nullobject** [comment]
Oh, I see it. Thanks for reporting this :eyes:

**Comment by 80sArcadeKid** [comment]
I also have a video of this glitch and comments in the MiSTer Forum.

Only happen on rounds 1 to 3, most prominent on round 1.
See post for more details, images, video.

**Comment by 80sArcadeKid** [comment]
Video Link in case you didn't get to the FPGA forum thread.
https://youtu.be/-axfq7Vd0Fw

I noticed another place the 'sprite sync' issue also shows up is when you are traversing up/down tunnels where the blue guys send out their tongues to hit you.  They shift up and down as you go up/down the rope. (i.e. rounds 5 and 9)

Knowing the game back to front in terms of gameplay, I think once these issues are resolved the game would play the same as my arcade PCB.

**Comment by ercan1441** [comment]
@nullobject any chance that this can be fixed soon? I think lot of Rygar fans will really appreciate this. I appreciate that there is a rygar core at all, but a stable core will be appreciated even more ;-)

**Comment by ercan1441** [comment]
@nullobject any chance that this can be fixed soon? I think lot of Rygar fans will really appreciate this. I appreciate that there is a rygar core at all, but a stable core will be appreciated even more ;-)

**Comment by 80sArcadeKid** [comment]
@nullobject yes would be great if this sprite sync issue could be fixed then the core would be perfect. :)
I notice that the issue almost dissapears after round 3 (level 3) but in the tunnels there is still issue with the 'licker' not quite syncing.

I also have a COMPLETE playthrough if that helps to look at any part of the game.
https://www.twitch.tv/videos/1226598434

From the start to 3:30 min you can see the dragons and 'death' head sprites are out of sync.
At 8:00 min the death head sprites on Round 3 is desynced.
At 10:00 mins (end of round 4) in the 'death head' comes in and is no longer out of sync.
At 11:38 mins (in the tunnels) you see as there is fast movement up and down on the rope that the 'lickers' are out of sync with the tunnel a bit.

Thats it! Thanks again for any updates.

**Comment by birdybro** [comment]
@nullobject was this fixed in the openFPGA version? I see you have rewritten it in scala there. I'm looking into seeing if I can fix this here. Any pointers?

**Comment by niknakniknak** [comment]
tested, no noticeable splitting

**Comment by Zet-sensei** [comment]
On Rygar / Argos no Senshi, the 4 first levels now seem to work correctly with the shared core from @paulb-nl.

The ticket should be closed @ercan1441 .

**Comment by Zet-sensei** [comment]
The code is pushed on the main branch & the core is updated, so we could close the issue this time.
